overlord/devicestate/devicemgr.go

```diff
@@ -599,6 +615,52 @@ func (m *DeviceManager) ensureOperational() error {
 	return nil
 }
 
+// maybeRestoreAfterReset attempts to restore the serial assertion with a
+// matching key in a post-factory reset scenario. It is possible that it is
+// called when the device was unregistered, but when doing so, the device key is
+// removed.
+func (m *DeviceManager) maybeRestoreAfterReset(device *auth.DeviceState) (keyID, serial string, err error) {
+	// there should be a serial assertion for the current model
+	serials, err := assertstate.DB(m.state).FindMany(asserts.SerialType, map[string]string{
+		"brand-id": device.Brand,
+		"model":    device.Model,
+	})
+	if (err != nil && asserts.IsNotFound(err)) || len(serials) == 0 {
+		// no serial assertion
+		return "", "", nil
+	}
+	for _, serial := range serials {
+		serialAs, ok := serial.(*asserts.Serial)
```

Comment:
we don't need the ok check here. we don't have it in other places. It's a serious bug if FindMany returns the wrong type of assertions